source.go

```diff
@@ -20,12 +20,36 @@ var (
 	_ (Source) = (*MultipleSource)(nil)
 )
 
+type SourceOptFunc func(*sourceOpt)
+
+type sourceOpt struct {
+	// onSubscriptionReceived is called when a subscription is received.
+	onSubscriptionReceived func(msg *nats.Msg)
```

Comment:
Nitpick(naming): we're receiving a message based on a subscription, not the subscription itself. `onMessageReceived` might be more appropriate here, and would align nicely with `onMessageRelayed`.

Reply:
agreed, renamed.